proposals/low-level-struct-improvements.md

```diff
@@ -805,25 +730,119 @@ For example the last line of calculating *safe-to-escape* of returns will change
 
 > the safe-to-escape of all argument expressions including the receiver. **This will exclude all arguments that line up with parameters marked as [DoesNotEscape]**
 
-Misc Notes:
-- The  `DoesNotEscapeAttribute` will be defined in the `System.Runtime.CompilerServices` namespace.
-- The `DoesNotEscapeAttribute` cannot be combined with the `[ThisRefEscapes]` attribute, doing so results in an error.
-- The `DoesNotEscapeAttribute` will be emitted as a `modreq`
+Detailed Notes:
+- The `[DoesNotEscape]` and `[RefThisEscapes]` cannot be combined on the same method 
+
+### Safe fixed size buffers
+The language will relax the restrictions on fixed sized arrays such that they can be declared in safe code and the element type can be managed or unmanaged.  This will make types like the following legal:
+
+```c#
+internal struct CharBuffer
+{
+    internal char Data[128];
+}
+```
+
+These declarations, much like their `unsafe` counter parts, will define a sequence of `N` elements in the containing type. These members can be accessed with an indexer and can also be converted to `Span<T>` and `ReadOnlySpan<T>` instances.
+
+When indexing into a `fixed` buffer of type `T` the `readonly` state of the container must be taken into account.  If the container is `readonly` then the indexer returns `ref readonly T` else it returns `ref T`. 
+
+Accessing a `fixed` buffer without an indexer has no natural type however it is convertible to `Span<T>` types. In the case the container is `readonly` the buffer is implicitly convertible to `ReadOnlySpan<T>`, else it can implicitly convert to `Span<T>` or `ReadOnlySpan<T>` (the `Span<T>` conversion is considered *better*). 
```

Comment:
> the `Span<T>` conversion is considered *better*

Why is this true?

Reply:
I would assume because the data is mutable so `Span<T>` is a better fit.

Plus, if you get a `Span<T>`, you can explicitly convert to `ROSpan<T>`. You cannot trivially do the inverse (and the inverse is generally considered "unsafe")

Reply:
Indeed that is the reason.